ddtrace/internal/ci_visibility/_api_client.py

```diff
@@ -0,0 +1,524 @@
+
+    if count_unparsed_suites:
+        log.warning("Failed to parse %d skippable suites", count_unparsed_suites)
+
+    record_skippable_count(len(suites_to_skip), SUITE)
+
+    return suites_to_skip
+
+
+def _parse_skippable_tests(
+    skippable_tests_data: t.List[_SkippableResponseDataItem], ignore_parameters: bool = False
+) -> t.Set[_SKIPPABLE_ITEM_ID_TYPE]:
+    tests_to_skip: t.Set[_SKIPPABLE_ITEM_ID_TYPE] = set()
+    count_unparsed_tests = 0
+    for skippable_test in skippable_tests_data:
+        try:
+            test_id = _get_test_id_from_skippable_test(skippable_test, ignore_parameters)
+            tests_to_skip.add(test_id)
+        except Exception:  # noqa: E722
+            log.warning("Failed to parse skippable test: %s", skippable_test, exc_info=True)
+            count_unparsed_tests += 1
+
+    if count_unparsed_tests:
+        log.warning("Failed to parse %d skippable tests", count_unparsed_tests)
+
+    record_skippable_count(len(tests_to_skip), TEST)
+
+    return tests_to_skip
+
+
+class _TestVisibilityAPIClientBase(abc.ABC):
+    """Client for fetching test visibility settings from the CI Visibility API
+
+    This class makes no direct references to environment variables, configs, or settings not passed to its
+    constructor (except for default values).
+    """
+
+    _requests_mode: REQUESTS_MODE
+
+    def __init__(
+        self,
+        base_url: str,
+        itr_skipping_level: ITR_SKIPPING_LEVEL,
+        git_data: GitData,
+        configurations: t.Dict[str, t.Any],
+        dd_service: t.Optional[str] = None,
+        dd_env: t.Optional[str] = None,
+        timeout: t.Optional[float] = None,
+    ):
+        self._base_url: str = base_url
+        self._itr_skipping_level: ITR_SKIPPING_LEVEL = itr_skipping_level
+        self._git_data: GitData = git_data
+        self._configurations: _CONFIGURATIONS_TYPE = configurations
+        self._service: t.Optional[str] = dd_service
+        self._dd_env: t.Optional[str] = dd_env
+        self._timeout: float = timeout if timeout is not None else DEFAULT_TIMEOUT
+
+    @abc.abstractmethod
+    def _redact_headers(self) -> t.Dict[str, str]:
+        """This is an abstract method to force child classes to consider which headers should be redacted for logging"""
+        pass
+
+    @abc.abstractmethod
+    def _get_headers(self) -> t.Dict[str, str]:
+        pass
+
+    def _get_final_headers(self) -> t.Dict[str, str]:
+        headers = _BASE_HEADERS.copy()
+        headers.update(self._get_headers())
+        return headers
+
+    def _do_request(self, method: str, endpoint: str, payload: str, timeout: t.Optional[float] = None) -> Response:
+        timeout = timeout if timeout is not None else self._timeout
+        headers = self._get_final_headers()
+        url = combine_url_path(self._base_url, endpoint)
+
+        conn: t.Optional[t.Union[HTTPSConnection | HTTPConnection | UDSHTTPConnection]] = None
```

Comment:
Isn't `|` the same as `t.Union`?

Reply:
Yeah, this was copy/paste'd from the original, but what I really want is `ddtrace.internal.utils.http.ConnectionType` anyway. Will update.